extension/storage/filestorage/client.go

```diff
@@ -172,7 +172,7 @@ func (c *fileStorageClient) Compact(compactionDirectory string, timeout time.Dur
 	}()
 
 	// use temporary file as compaction target
-	options := bboltOptions(timeout, false)
+	options := bboltOptions(timeout, !c.db.NoSync)
```

Comment:
Thanks for fixing this.

Could I ask a minor related item, that we invert the rename the `fSync` parameter we're passing around to `noSync` and invert the usage? (within the functions but also Also requires inverting the value [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3cb7d34ad7f88fc864d1d064e82d0769270cf946/extension/storage/filestorage/extension.go#L67).) This would allow us to avoid the double negative here.

Reply:
Yeah works for me. 

Do you also want to change the config option? I think it's best to have fsync on by default unless you _really_ know what you're doing, since it can cause data loss and crash looping of otelcol. But maybe that should be a separate PR

Reply:
If you don't mind, please open a new issue to propose the change.